### PR TITLE
REST API endpoint for listing deliveries

### PR DESCRIPTION
**Pull request by alexeyvy**
According to https://oat-sa.atlassian.net/browse/TAO-6725

### CODE REVIEW

File: controller/RestDelivery.php [hunk 1]
```diff
@@ -254,6 +254,59 @@ public function deleteDeferred()
...
+            if ($this->hasRequestParameter('limit')) {
+                $limit = (int)$this->getRequestParameter('limit');
+                if ($limit < 0) {
+                    throw new \common_exception_ValidationFailed('limit', '\'Limit\' should be a positive integer');
+                }
+            }
+
+            $offset = 0;
+            if ($this->hasRequestParameter('offset')) {
+                $offset = (int)$this->getRequestParameter('offset');
```
[the remark this answers is not in the document]
Comment (chilly86) [reply controller/RestDelivery.php]: remove the (int) ;)

[the remark this answers is not in the document]
Comment (alexeyvy) [reply controller/RestDelivery.php]: Ooops sorry! Done

File: controller/RestDelivery.php [hunk 2]
```diff
@@ -254,6 +254,59 @@ public function deleteDeferred()
...
+                    throw new \common_exception_ValidationFailed('offset', '\'Offset\' should be a positive integer');
+                }
+            }
+
+            $service = DeliveryAssemblyService::singleton();
+
+            /** @var \core_kernel_classes_Resource[] $deliveries */
+            $deliveries = $service->getAllAssemblies();
+            $overallCount = count($deliveries);
+            if ($offset || $limit) {
```
[the remark this answers is not in the document]
Comment (chilly86) [reply controller/RestDelivery.php]: yeah I mixed up offset with page :)

File: controller/RestDelivery.php [hunk 3]
```diff
@@ -254,6 +254,62 @@ public function deleteDeferred()
...
+                $limit = $this->getRequestParameter('limit');
+                if (!is_numeric($limit) || $limit < 0) {
+                    throw new \common_exception_ValidationFailed('limit', '\'Limit\' should be a positive integer');
+                }
+            }
+
+            $offset = 0;
+            if ($this->hasRequestParameter('offset')) {
+                $offset = $this->getRequestParameter('offset');
+                if (!is_numeric($offset) || $offset < 0) {
```
Comment (chilly86) [inline controller/RestDelivery.php]: To make sure it's not a float
if (!is_numeric($offset) || intval($offset) != $offset || $offset < 0) {

File: controller/RestDelivery.php [hunk 4]
```diff
@@ -254,6 +254,62 @@ public function deleteDeferred()
...
+    {
+        try {
+            if ($this->getRequestMethod() !== \Request::HTTP_GET) {
+                throw new \common_exception_NotImplemented('Only get method is accepted to getting deliveries');
+            }
+
+            $limit = 0;
+            if ($this->hasRequestParameter('limit')) {
+                $limit = $this->getRequestParameter('limit');
+                if (!is_numeric($limit) || $limit < 0) {
```
Comment (chilly86) [inline controller/RestDelivery.php]: To make sure it's not a float
if (!is_numeric($offset) || intval($offset) != $offset || $offset < 0) {

File: controller/RestDelivery.php (hunk 3 above)
Comment (alexeyvy) [reply controller/RestDelivery.php]: Done :)

**Pull request merged**